migtests/scripts/functions.sh

```diff
@@ -1472,6 +1472,80 @@ generate_voyager_config() {
 	fi
 }
 
+normalize_callhome_json() {
+    local input_file="$1"
+    local output_file="$2"
+    local temp_file="/tmp/temp_file.json"
+
+    # Normalize JSON with jq; use --sort-keys to avoid the need to keep the same sequence of keys in expected vs actual json
+    jq --sort-keys 'walk(
+        if type == "object" then
+            .ObjectNames? |= (
+				if type == "string" then
+					split(", ") | sort | join(", ")
+				else
+					.
+				end
+			) |
+            .VoyagerVersion? = "IGNORED" |
+			.TargetDBVersion? = "IGNORED" |
+            .OptimalSelectConnectionsPerNode? = "IGNORED" |
+            .OptimalInsertConnectionsPerNode? = "IGNORED" |
+			.SizeInBytes? = "IGNORED" |
+            .yb_cluster_metrics = "IGNORED" |
+            .parallel_jobs = "IGNORED" |
+            .adaptive_parallelism_max = "IGNORED" |
+            .snapshot_total_bytes = "IGNORED"
+        elif type == "array" then
+			sort_by(tostring)
+        elif type == "string" and (
+            # Check if the string is a JSON array and sort it
+            (tostring | test("^ *?\\[.*\\] *?$")) and (fromjson | type == "array")
+        ) then
+            fromjson | sort_by(tostring) | tojson
+		else
+            .
+        end
+    )' "$input_file" > "$temp_file"
+
+    # Move cleaned file to output
+    mv "$temp_file" "$output_file"
+}
+
+compare_callhome_json_reports() {
+    local expected_report_file="$1"
+    local phase="$2"
+    
+    # Get report data from API and save to local file
+    local actual_report_file=$(mktemp)
+    if curl -s -o "$actual_report_file" "http://localhost:5000/get_payload/$phase"; then
+        echo "Successfully retrieved report data from API"
+        echo "Report data saved to: $actual_report_file"
+    else
+        echo "Failed to retrieve report data from API"
+        rm -f "$actual_report_file"
```

Comment:
What are we removing if the report retrieval failed? Also is it possible to get some error / context here as to why this failed?